src/state/routing/useRoutingAPITrade.ts

```diff
@@ -0,0 +1,175 @@
+import { skipToken } from '@reduxjs/toolkit/query/react'
+import { Currency, CurrencyAmount, TradeType } from '@uniswap/sdk-core'
+import { Trade } from '@uniswap/v3-sdk'
+import { BigNumber } from 'ethers'
+import ms from 'ms.macro'
+import { useMemo } from 'react'
+import { useBlockNumber } from 'state/application/hooks'
+import { useGetQuoteQuery } from 'state/routing/slice'
+import { V3TradeState } from './types'
+import { computeRoutes } from './computeRoutes'
+
+function useFreshData<T>(data: T, dataBlockNumber: number, maxBlockAge = 10): T | undefined {
+  const localBlockNumber = useBlockNumber()
+
+  if (!localBlockNumber) return undefined
+  if (localBlockNumber - dataBlockNumber > maxBlockAge) {
+    return undefined
+  }
+
+  return data
+}
+
+/**
+ * Returns query arguments for the Routing API query or undefined if the
+ * query should be skipped.
+ */
+function useRoutingAPIArguments({
+  tokenIn,
+  tokenOut,
+  amount,
+  type,
+}: {
+  tokenIn: Currency | undefined
+  tokenOut: Currency | undefined
+  amount: CurrencyAmount<Currency> | undefined
+  type: 'exactIn' | 'exactOut'
+}) {
+  if (!tokenIn || !tokenOut || !amount || tokenIn.equals(tokenOut)) {
+    return undefined
+  }
+
+  return {
+    tokenInAddress: tokenIn.wrapped.address,
+    tokenInChainId: tokenIn.chainId,
+    tokenOutAddress: tokenOut.wrapped.address,
```

Comment:
hmm does the api not accept ETH as some kind of argument?

Reply:
it does accept `ETH` as `tokenAddress` but @willpote mentioned passing WETH will not change the outcome

Reply:
true, we do have `currencyId` function though and may want to be accurate if it doesn't cost us anything